### DISCUSSION TITLE
Quarkus Websockets

### DISCUSSION THREAD

**Discussion by phillip-kruger** [quarkusio/quarkus#35520] [Community]
/cc @maxandersen @cescoffier 
We are looking at the websockets support in Quarkus and how we can improve it. This discussion will be used to discuss idea and tasks related to this.

### Notes: 
- No dependency on Servlet as thought initially.
- Executing on the eventloop, not on the worker thread as thought initially

### Features we maybe want ?

- [ ] Quarkus-websocket-jackson and quarkus-websocket-jsonb (and maybe others like xml ?) - same as RestEasy ?
- [ ] JsonRPC support / extension 
- [ ] Auto session management
- [ ] Broadcast and reply feature
- [ ] Object binding (json)
- [ ] Methods with Uni for non blocking reply methods
- [ ] Methods without Uni for blocking reply methods
- [ ] Methods with Multi for broadcast ?
- [ ] Security ?
- [ ] Duplicated context support
- [ ] Virtual thread support 
- [ ] Kotlin co-routines support.
- [ ] ...

**Comment by mschorsch** [comment]
Kotlin coroutines support.
